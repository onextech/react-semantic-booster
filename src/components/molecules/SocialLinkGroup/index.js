import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Button, Icon } from 'semantic-ui-react';
import { kebabCase } from 'lodash';


const socialLinkGroupClassName = 'social-link-group';
const borderlessClassName = 'borderless';

const StyledButtonGroup = styled(Button.Group)`
  &.ui.buttons.${socialLinkGroupClassName} {
  
    &:not(.basic) {
      .button {
        border-radius: 0;
      }
    }
  
    // borderless
    &.borderless {
      border: none;
      .button {
        &, &:hover, &:active, &:focus {
          box-shadow: none;
        }
      }
    }
    
  }
`;

const TextSpan = styled.span`
  margin-left: .25em;
  margin-right: .4em;
`;

const SocialLinkGroup = ({ items, groupProps }) => {
  const getClassName = () => {
    const className = [socialLinkGroupClassName];
    if (groupProps && groupProps.borderless) {
      className.push(borderlessClassName);
    }
    return className.join(' ');
  };

  // separate custom from semantic props
  let semanticButtonGroupProps = {};
  if (groupProps) {
    const { borderless, ...rest } = groupProps;
    semanticButtonGroupProps = rest;
  }

  return (
    <StyledButtonGroup className={getClassName()} icon {...semanticButtonGroupProps}>
      {
        items.map(item => <a {...item.link} key={kebabCase(item.name)}>
          <Button {...item.btn}>
            <Icon {...item.icon} />
            {item.text && <TextSpan>{item.text}</TextSpan>}
          </Button>
        </a>)
      }
    </StyledButtonGroup>
  );
};

SocialLinkGroup.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    link: PropTypes.object, // <a> props
    btn: PropTypes.object, // semantic-ui button props
    icon: PropTypes.object, // semantic-ui icon props
    text: PropTypes.string,
  })).isRequired,
  groupProps: PropTypes.shape({
    borderless: PropTypes.bool, // custom: only works when basic: true
    // ... others, semantic-ui button group props
  }),
};

export default SocialLinkGroup;
