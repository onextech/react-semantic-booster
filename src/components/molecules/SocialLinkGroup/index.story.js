import React from 'react';
import { storiesOf } from '@storybook/react';
import SocialLinkGroup from '.';


export const itemsExample = [
  {
    name: 'facebook',
    link: { href: 'http://www.facebook.com', target: '_blank' },
    icon: { name: 'facebook f' },
    btn: { color: 'facebook' },
  },
  {
    name: 'instagram',
    link: { href: 'http://www.instagram.com', target: '_blank' },
    icon: { name: 'instagram' },
    btn: { color: 'instagram' },
  },
  {
    name: 'twitter',
    link: { href: 'http://www.twitter.com', target: '_blank' },
    icon: { name: 'twitter' },
    btn: { color: 'twitter' },
  },
];

export const groupPropsExample = { size: 'massive' };

storiesOf('SocialLinkGroup', module)
  .add('Default', () => (
    <SocialLinkGroup items={itemsExample} groupProps={{ ...groupPropsExample }} />
  ))
  .add('Basic', () => (
    <SocialLinkGroup
      items={itemsExample}
      groupProps={{ ...groupPropsExample, basic: true, borderless: true }} />
  ));
