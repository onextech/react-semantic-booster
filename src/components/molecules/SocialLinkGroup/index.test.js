import React from 'react';
import renderer from 'react-test-renderer';
import SocialLinkGroup from '.';
import { itemsExample, groupPropsExample } from './index.story';


describe('<SocialLinkGroup>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SocialLinkGroup items={itemsExample} groupProps={groupPropsExample} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
