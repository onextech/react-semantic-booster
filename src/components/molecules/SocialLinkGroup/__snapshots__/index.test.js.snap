// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLinkGroup> renders correctly 1`] = `
<div
  className="ui massive icon buttons social-link-group SocialLinkGroup__StyledButtonGroup-iaUnjP bPLyMQ"
>
  <a
    href="http://www.facebook.com"
    target="_blank"
  >
    <button
      className="ui facebook button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="facebook f icon"
      />
    </button>
  </a>
  <a
    href="http://www.instagram.com"
    target="_blank"
  >
    <button
      className="ui instagram button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="instagram icon"
      />
    </button>
  </a>
  <a
    href="http://www.twitter.com"
    target="_blank"
  >
    <button
      className="ui twitter button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="twitter icon"
      />
    </button>
  </a>
</div>
`;
